src/actionHandlers/commands/warn.ts

```diff
@@ -0,0 +1,112 @@
+import { ApplicationCommandOptionType, EmbedBuilder, userMention } from 'discord.js';
+
+import type { Command } from './types';
+import { getUserItem, setUserSchemaItem } from '../../db';
+import { onAutoBan } from '../onAutoBan';
+
+export const warn: Command = async (interaction) => {
+  const warnItem = interaction.options.getSubcommand();
+  const user = interaction.options.getUser('user').id;
+  const reason = interaction.options.getString('reason');
+  const warnsCount = (await getUserItem(interaction.guildId, user, 'warns')).number;
+  const warnReasons = (await getUserItem(interaction.guildId, user, 'warns')).reasons;
```

Comment:
Separate getUserItem Function to a variable.
you don't need to separate warns count and warns reasons into two variable, just access them from the userWarns variable that u will create